### ISSUE TITLE
Execute docker complex command lines (with more than one line)

### ISSUE THREAD

**Issue by varunshankar** [varunshankar/galaxy-godocker_tool#3]
The user can only write one line. We have to think about the best manner to give opportunity to write a multi-line complex command.For now, we can use | or something like that but maybe not the best manner.

**Comment by varunshankar** [comment]
Hi,
I have updated the code. I considered the commands to be executed is sequential , so I used the ';' instead of '|' (pipeline) operator.

**Comment by osallou** [comment]
if input is a textarea, it can be used as input with the --script option of the godocker CLI if you use the CLI. If you use API, then it matches the command parameter of the job.
You just need to define your command input in galaxy to be a textarea instead of a input (don't know what is the type for galaxy)

**Comment by yvanlebras** [comment]
Hi Varun,
For a text area parameter in Galaxy, you can try this kind of command line on the tool xml:
<param name="inputarea" type="text" area="True" size="5x25" /> 
Cheers,
Yvan 

Envoyé depuis mon appareil Samsung

-------- Message d'origine --------
De : Olivier Sallou notifications@github.com 
Date : 20/08/2016  22:51  (GMT+01:00) 
À : varunshankar/galaxy-godocker_tool galaxy-godocker_tool@noreply.github.com 
Cc : Yvan Le Bras yvan.le_bras@irisa.fr, Assign assign@noreply.github.com 
Objet : Re: [varunshankar/galaxy-godocker_tool] Execute docker complex command lines (with more than one line) (#3) 

if input is a textarea, it can be used as input with the --script option of the godocker CLI if you use the CLI. If you use API, then it matches the command parameter of the job.

You just need to define your command input in galaxy to be a textarea instead of a input (don't know what is the type for galaxy)

—
You are receiving this because you were assigned.
Reply to this email directly, view it on GitHub, or mute the thread.

{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/varunshankar/galaxy-godocker_tool","title":"varunshankar/galaxy-godocker_tool","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/varunshankar/galaxy-godocker_tool"}},"updates":{"snippets":[{"icon":"PERSON","message":"@osallou in #3: if input is a textarea, it can be used as input with the --script option of the godocker CLI if you use the CLI. If you use API, then it matches the command parameter of the job.\r\nYou just need to define your command input in galaxy to be a textarea instead of a input (don't know what is the type for galaxy)"}],"action":{"name":"View Issue","url":"https://github.com/varunshankar/galaxy-godocker_tool/issues/3#issuecomment-241223030"}}}​​​

**Comment by varunshankar** [comment]
Ya... textarea is the right one! :) Its working well if we have commands separated with semicolons. eg an input in the text area like the following one works good:

a=10;
b=20;
val=`expr $a + $b`;
echo "a + b : $val";
val=`expr $a - $b`;
echo "a - b : $val";
echo "hello world";

**Comment by osallou** [comment]
Carriage return in textarea should be enough, semicolons are not necessary

Le lun. 22 août 2016 23:39, Varun Shankar S notifications@github.com a
écrit :

**Comment by varunshankar** [comment]
I have also checked with only carriage return and it is not working. I can process the textarea and insert semicolons in the back end.

**Comment by osallou** [comment]
Textarea content is converted in single line? Don t think so

Le mar. 23 août 2016 06:59, Varun Shankar S notifications@github.com a
écrit :

**Comment by varunshankar** [comment]
Yes... it is getting converted into a single line when making script file out of the text area input
